raster-foundry/raster-foundry
# Add Caching and Various Fixes

## Overview

Adds configurable caching for backsplash for the following parts:
 - tiles
 - authorization
 - histograms
 - authentication
 - raster sources

Also fixes multiband mosaics for RGB visualization and fixes checking for nodata values.

### Checklist


### Demo

```
backsplash_1        | 20:42:52.237 [main] INFO  com.rasterfoundry.backsplash.server.Cache$ - Authorization Cache Status (read/write) Flags(true,true)
backsplash_1        | 20:42:52.237 [main] INFO  com.rasterfoundry.backsplash.server.Cache$ - Authentication Cache Status: Flags(true,true)
backsplash_1        | 20:42:52.250 [main] INFO  com.rasterfoundry.backsplash.Cache$ - Tile Cache Status: Flags(true,true)
backsplash_1        | 20:42:52.250 [main] INFO  com.rasterfoundry.backsplash.Cache$ - Histogram Cache Status: Flags(true,true)
backsplash_1        | 20:42:52.250 [main] INFO  com.rasterfoundry.backsplash.Cache$ - Raster Source Cache Status: Flags(true,true)
backsplash_1        | 20:42:53.793 [main] INFO  com.rasterfoundry.backsplash.server.Main$ - Server Started (1)
backsplash_1        | 20:42:53.924 [http4s-0] INFO  o.http4s.blaze.channel.nio1.NIO1SocketServerGroup - Service bound to address /0.0.0.0:8080
backsplash_1        | 20:42:53.927 [http4s-0] INFO  org.http4s.server.blaze.BlazeServerBuilder - 
backsplash_1        |   _   _   _        _ _
backsplash_1        |  | |_| |_| |_ _ __| | | ___
backsplash_1        |  | ' \  _|  _| '_ \_  _(_-<
backsplash_1        |  |_||_\__|\__| .__/ |_|/__/
backsplash_1        |              |_|
backsplash_1        | 20:42:53.993 [http4s-0] INFO  org.http4s.server.blaze.BlazeServerBuilder - http4s v0.20.0-M4 on blaze v0.14.0-M11 started at http://0.0.0.0:8080/
```

## Testing Instructions

 * Rebuild backsplash server (`./scripts/console sbt` and then `backsplash-server/assembly`)
 * Start backsplash, adjust cache settings in `docker-compose.java.yml` between sessions to see performance)

Closes #4427 for RGB Mosaics
